### PR TITLE
Store to regions

### PR DESCRIPTION
**Pull request by laliberte**
This enhancement covers two use cases:

1. ``da.store(a, b, regions=setitem_tuple)`` is the same as ``b[setitem_tuple] = a``.

2. ``da.store([a, c], [b, d], regions=[setitem_tuple1, setitem_tuple2])`` is the same as ``b[setitem_tuple1] = a`` and ``d[setitem_tuple2] = c``.

### CODE REVIEW

**Comment by mrocklin** [comment]
At first glance this seems like a good approach to me.

cc @shoyer

File: dask/array/core.py [hunk 1]
```diff
@@ -787,8 +791,18 @@ def store(sources, targets, lock=True, compute=True, **kwargs):
         raise ValueError("Different number of sources [%d] and targets [%d]"
                          % (len(sources), len(targets)))
 
-    updates = [insert_to_ooc(tgt, src, lock=lock)
-               for tgt, src in zip(targets, sources)]
+    if isinstance(regions, tuple) or regions is None:
+        regions = [regions]
+
+    if len(sources) > 1 and len(regions) == 1:
+        regions = [regions[0] for src in sources]
```
Comment (pitrou) [inline dask/array/core.py]: A clearer way to express this is `regions = regions * len(source)` or even `regions *= len(sources)`

File: dask/array/tests/test_array_core.py [hunk 2]
```diff
@@ -928,6 +928,34 @@ def test_store():
...
+def test_store_regions():
+    d = da.ones((4, 4, 4), chunks=(2, 2, 2))
+    a, b = d + 1, d + 2
+
+    at = np.zeros(shape=(8, 4, 6))
+    bt = np.zeros(shape=(8, 4, 6))
+
+    region = (slice(None,None,2), slice(None), [1, 2, 4, 5])
+
+    #Single region:
```
Comment (shoyer) [inline dask/array/tests/test_array_core.py]: style: always include a space after `#` in comments

**Comment by shoyer** [comment]
I have one style nit, but otherwise looks good to me.

File: dask/array/tests/test_array_core.py (hunk 2 above)
Comment (laliberte) [inline dask/array/tests/test_array_core.py]: Modified in new comments. Should I have remembered from our previous interaction with h5netcdf. Thanks ;)

**Pull request merged**